tiledb/sm/c_api/tiledb.cc

```diff
@@ -2341,6 +2341,68 @@ int32_t tiledb_array_schema_load_with_key(
   return TILEDB_OK;
 }
 
+int32_t tiledb_array_remove_attribute(
```

Comment:
We try to keep the amount of logic in the c-api to a minimum. This function is a good candidate to move into the storage manager to encapsulate in the C++ library. I'm sure you modeled this after the `tiledb_create_array` functionality where we have significant logic. The `tiledb_create_array` is unfortunately a bad example and itself should be adjusted to move the logic into C++ and the c-api becomes thing callers into the storage manager there too. That's for another PR.

Reply:
I will move this and tiledb_array_add_attribute(in another branch) to storage_manager

Reply:
the main logic is moved to the storage manager.